lib/logstash/codecs/multiline.rb

```diff
@@ -111,6 +115,13 @@ module LogStash module Codecs class Multiline < LogStash::Codecs::Base
   # This only affects "plain" format logs since JSON is `UTF-8` already.
   config :charset, :validate => ::Encoding.name_list, :default => "UTF-8"
 
+  # Change the delimiter that separates lines
+  config :delimiter, :validate => :string, :default => "\n"
+
+  # Assume data received from input plugin as line based. For some input plugins
+  # like stdin or tcp/udp data is not line based and this option should be set to false.
+  config :line_based_input, :validate => :boolean, :default  => true
```

Comment:
but that would need us to properly annotate all input plugins? in which case this option allows us to  have something working today, and eventually we can deprecate this once we have the proper info  from the input plugins?